### DISCUSSION TITLE
Support concurrent / parallel workspace command executions

### DISCUSSION THREAD

**Discussion by dreamorosi** [npm/feedback#781] [CLI]
With this discussion I'd like to suggest and improvement to the npm workspaces feature.

At the moment npm allows you to run commands like `npm t -ws` which would run `npm t` in each one of the workspace packages of said workspace.

This is great, however the commands are run sequentially and at the best of my knowledge, it seems that the order in which they are listed in `package.json` is respected.

It would be great if for selected commands/situations, the CLI supported a flag like `--concurrent` (or similar name) that would tell `npm` to instead run these commands concurrently and not one after the other.

At the moment, if you want to achieve the same result, you have to manually run `npm t -w package/a & npm t -w package/b` which works, but is not ideal.

I have tried to search for similar discussions in this repo but couldn't find anything using the keywords "concurrent", "parallel", or "sequential". If there's already another discussion on this topic please feel free to close this in favor of the original one.

**Comment by ljharb** [comment]
See:
 - https://github.com/npm/rfcs/issues/190
 - https://github.com/npm/rfcs/issues/610
 - https://github.com/npm/rfcs/issues/573

**Comment by itsjoekent** [comment]
I don't really understand why this was marked as answered. The threads listed above are in regards to running concurrent npm commands within a single npm project, this thread is requesting concurrent npm commands for workspaces. To the original authors point, it is logistically impossible to use workspaces natively as a development tool if I cannot run multiple "watch" commands at the same time.

**Reply by dreamorosi** [reply]
I marked it as answered because two of the three issues linked in the answer are RFCs for the feature I was looking for.

While I agree that the feature is not there yet and that it'd be great to have it, I think an RFC is a better place than a discussion to continue the conversation.
